test/support/utils_invoke.h

```diff
@@ -165,6 +177,89 @@ get_dpcpp_test_policy()
         throw;
     }
 }
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+// struct test_policy_container - a container for policy which return saved policy
+// as l-value or r-value depends on source policy type qualifiers
+template <typename _PolicySource, typename _PolicyNew>
+struct test_policy_container
+{
+    using _PolicyNewDecayed = std::decay_t<_PolicyNew>;
+
+    _PolicyNewDecayed __policy_new;
+
+    test_policy_container(_PolicyNewDecayed&& __policy_new)
+        : __policy_new(std::move(__policy_new))
+    {
+    }
+
+    // Delete copy constructor to avoid copying of test_policy_container
+    test_policy_container(const test_policy_container&) = delete;
+    test_policy_container(test_policy_container&&) = delete;
+
+    // Delete assignment operator to avoid copying of test_policy_container
+    test_policy_container& operator=(const test_policy_container&) = delete;
+    test_policy_container& operator=(test_policy_container&&) = delete;
+
+    // Return testing policy
+    decltype(auto) get()
+    {
+        using TestingPolicyType = std::conditional_t<
```

Comment:
suggestion: Is it cleaner to move this `TestingPolicyType` outside of the function and use it as the return type in `get()`?

Reply:
Done